apps/web/src/middlewares/ip-address-middleware.ts

```diff
@@ -0,0 +1,17 @@
+import { NextFetchEvent, NextMiddleware, NextRequest, NextResponse } from 'next/server'
+import { shouldGeoBlock } from '@pancakeswap/utils/geoBlock'
+import { MiddlewareFactory } from './types'
+
+export const withUserIpHeaders: MiddlewareFactory = (next: NextMiddleware) => {
+  return async (request: NextRequest, _next: NextFetchEvent) => {
+    let ip = request.ip ?? request.headers.get('x-real-ip')
+    const forwardedFor = request.headers.get('x-forwarded-for')
+
+    if (!ip && forwardedFor) ip = forwardedFor.split(',').at(0) ?? 'Unknown'
```

Comment:
Leave it as what it is if no valid ip is found

```suggestion
    if (!ip && forwardedFor) ip = forwardedFor.split(',').at(0) ?? ip
```